doc/proposals/scorecard-user-tests.md

```diff
@@ -0,0 +1,112 @@
+# User Defined Tests for the Operator Scorecard
+
+Implementation Owner: AlexNPavel
+
+Status: Draft
+
+[Background](#Background)
+
+[Goals](#Goals)
+
+[Design overview](#Design_overview)
+
+[User facing usage](#User_facing_usage)
+
+[Observations and open questions](#Observations_and_open_questions)
+
+## Background
+
+The operator scorecard is intended to allow users to run a generic set of tests on their operators. Some simple checks can be performed, but more complicated
+tests verifying that the operator actually works are not possible to do in a generic way. This leads to some of the functional tests in the current scorecard
+implementation to be inaccurate or too insufficient to be useful. For more useful functional scorecard tests, we need to allow some basic user input for tests.
+
+## Goals
+
+- Implement user-defined scorecard tests
+- Replace existing "Operator actions are reflected in status" and "Writing into CRs has effects" tests with user-defined tests
+
+## Design overview
+
+A new section would be added to the scorecard config file called `functional_tests` where a user can define various aspects of a test. For example, this definition runs a similar test to the memcached-operator scale test from the SDK's e2e test:
+
+```yaml
+functional_tests:
+  - cr: "deploy/crds/memcached.cr.yaml"
+    expected_resources:
```

Comment:
Not sure why the `expected_resources` and `expected_status` are separate, wdyt of having just `expected` and group all of it under that?

Reply:
That's explained a little bit lower. We use `expected_resources` to replace the "Writing into CRs has an effect" test (as it interacts with and modifies resources in the cluster) and `expected_status` replaces the `Operator actions are reflected in status` test. Having `expected_status` helps us make sure that the user is updating the status of their CR and also makes it a bit easier for the user to define the status block checks for the CR.

Reply:
We could change it to this if you prefer:

```yaml
expected:
  resources:
    ...
  status:
    ...
```

I just want it to be clear that updating the status of the CR and checking that the operator correctly configured other resources are individual tests.

Reply:
I think the above makes a bit more sense and is cleaner and separated. But no strong opinions on it, up to you. 👍